backend/corpora/common/entities/dataset.py

```diff
@@ -51,23 +42,20 @@ def create(
             sex=sex,
             ethnicity=ethnicity,
             development_stage=development_stage,
-            artifacts=cls._create_sub_objects(artifacts, DbDatasetArtifact, add_columns=dict(dataset_id=primary_key)),
-            deployment_directories=cls._create_sub_objects(
-                deployment_directories,
-                DbDeploymentDirectory,
-                add_columns=dict(dataset_id=primary_key),
-            ),
-            processing_status=cls._create_sub_object(
-                processing_status, DbDatasetProcessingStatus, add_columns=dict(dataset_id=primary_key)
-            ),
             **kwargs,
         )
-
-        cls.db.session.add(new_db_object)
-        cls.db.session.flush()
+        if artifacts:
```

Comment:
Not a big deal, but do you even need the `if artifacts` here?

Reply:
No we don't

Reply:
it turns out it is needed otherwise we might try iterating over None.